docs/src/pages/customization/palette/palette.md

```diff
@@ -209,6 +209,10 @@ The colors modified by the palette mode are the following:
 
 {{"demo": "pages/customization/palette/DarkTheme.js", "bg": "inline", "hideToolbar": true}}
 
+> Note: The colors are modified only if you use the default palette. If you have a custom palette, you need to make sure that you have the correct values based on the `mode`.
+
+{{"demo": "pages/customization/palette/DarkThemeWithCustomPalette.js", "defaultCodeOpen": false}}
```

Comment:
Why not add a demo, but I don't think that it's how we will solve most of the pain of #28076. Why? Because for somebody that doesn't know the library, it feels overwhelming. I have to pause to understand what happens in this demo. A first great step would be to make it as small as possible to retain the main point. I think that we need to guide the developers.

Also, I would personally close #28076 as a duplicate of #21757.

Reply:
> I think that we need to guide the developers more, using the demo as an illustration, not as the primary piece of information. We could add a detailed explanation, it would create context, make this information searchable.

Agree, will extend the explanation first and try to simplify the demo

> Also, I would personally close #28076 as a duplicate of #21757.

I think these two are different issues. The one is how to customize the theme based on the other, and the other is more of a why dark theme doesn't work when on the docs it said it should (because you have custom palette and you need to handle it by yourself).

Reply:
Updated, hopefully, is more clear now